crates/op-rbuilder/src/builders/flashblocks/payload.rs

```diff
@@ -453,12 +491,13 @@ where
                         Err(err) => {
                             // Track invalid/bad block
                             ctx.metrics.invalid_blocks_count.increment(1);
-                            error!(target: "payload_builder", "Failed to build block {}, flashblock {}: {}", ctx.block_number(), flashblock_count, err);
+                            error!(target: "payload_builder", "Failed to build block {}, flashblock {}: {}", ctx.block_number(), ctx.flashblock_index(), err);
                             // Return the error
                             return Err(err);
                         }
                         Ok((new_payload, mut fb_payload, new_bundle_state)) => {
-                            fb_payload.index = flashblock_count + 1; // we do this because the fallback block is index 0
+                            ctx.increment_flashblock_index(); // fallback block is index 0, so we need to increment here
+                            fb_payload.index = ctx.flashblock_index();
```

Comment:
let's add function that return incremented index and use it in here?
I feel like using increment_flashblock_index + flashblock_index is very brittle and we could accidentally remove/forgot something